pkg/pipeline/WorkflowDagExecutor.go

```diff
@@ -2390,7 +2390,7 @@ func (impl *WorkflowDagExecutorImpl) ManualCdTrigger(overrideRequest *bean.Value
 			return 0, err
 		}
 		// Migration of deprecated DataSource Type
-		if artifact.DataSource == repository.EXT {
+		if artifact.IsMigrationRequired() {
```

Comment:
instead of removing this check we can _if artifact.DataSource == repository.EXT_ inside _artifact.IsMigrationRequired()_ and call _impl.ciArtifactRepository.MigrateToWebHookDataSourceType(artifact.Id)_ inside  _if artifact.DataSource == repository.EXT_

Reply:
but we need to migrate all invalid data_source types to webhook